ifacemonitor/ifacemonitor_test.go

```diff
@@ -179,12 +196,15 @@ func (nl *netlinkTest) LinkList() ([]netlink.Link, error) {
 			},
 		})
 	}
+	nl.linksMutex.Unlock()
 	return links, nil
 }
 
 func (nl *netlinkTest) AddrList(link netlink.Link, family int) ([]netlink.Addr, error) {
 	name := link.Attrs().Name
+	nl.linksMutex.Lock()
 	model, prs := nl.links[name]
+	nl.linksMutex.Unlock()
```

Comment:
I think the loop below needs to be guarded by the mutex too.  `addAddr()` could be updating the set that you're iterating over here.

(To be clear, I'm not sure if that can actually happen in the scope of the test, but it's easier/more robust to fix than it is to reason about the large-scale behaviour of the test.)